### PR TITLE
Add support for fastpath anomaly/confirmed/failure cols

### PR DESCRIPTION
**Pull request by FedericoCeratto**
To be deployed only after https://github.com/ooni/pipeline/pull/271

### CODE REVIEW

**Comment by hellais** [comment]
This may have to be dropped in light of the new changes to the data model

**Comment by hellais** [comment]
I am removing myself as a reviewer and I suspect we will have to close this PR. @FedericoCeratto I will leave that decision to you.

**Comment by FedericoCeratto** [comment]
The changes in fastpath in https://github.com/ooni/pipeline/pull/271 are deployed. Rebasing and updating this PR.

**Comment by FedericoCeratto** [comment]
@hellais the PR is now up to date (essentially rewritten from scratch)

File: measurements/api/measurements.py [hunk 1]
```diff
@@ -482,21 +469,34 @@ def list_measurements(
...
-        fpwhere.append(sql.text("false"))
+    # anomaly and confirmed: any value != TRUE is treated as FALSE
+
+    if anomaly is True:
+        mrwhere.append(sql.text("measurement.anomaly IS TRUE"))
+        fpwhere.append(sql.text("fastpath.anomaly IS TRUE"))
+
+    elif anomaly is False:
+        mrwhere.append(sql.text("measurement.anomaly IS NOT TRUE"))
+        fpwhere.append(sql.text("fastpath.anomaly IS NOT TRUE"))
```
Comment (hellais) [inline measurements/api/measurements.py]: Given the `anomaly` is nullable, saying `anomaly = False` is not the same as `anomaly != True`.

I suggest changing this to:
```
mrwhere.append(sql.text("measurement.anomaly IS FALSE"))
```

File: measurements/api/measurements.py [hunk 2]
```diff
@@ -482,21 +469,34 @@ def list_measurements(
...
+
+    if confirmed is True:
+        mrwhere.append(sql.text("measurement.confirmed IS TRUE"))
+        fpwhere.append(sql.text("fastpath.confirmed IS TRUE"))
+
+    elif confirmed is False:
+        mrwhere.append(sql.text("measurement.confirmed IS NOT TRUE"))
+        fpwhere.append(sql.text("fastpath.confirmed IS NOT TRUE"))
```
Comment (hellais) [inline measurements/api/measurements.py]: Same as above

File: measurements/api/measurements.py [hunk 3]
```diff
@@ -482,21 +469,34 @@ def list_measurements(
...
+
+    if failure is True:
+        # residual_no is never NULL, msm_failure is always NULL
+        mrwhere.append(sql.text("measurement.exc IS NOT NULL"))
+        fpwhere.append(sql.text("fastpath.msm_failure IS TRUE"))
+
+    elif failure is False:
+        # on success measurement.exc is NULL
+        mrwhere.append(sql.text("measurement.exc IS NULL"))
+        fpwhere.append(sql.text("fastpath.msm_failure IS NOT TRUE"))
```
Comment (hellais) [inline measurements/api/measurements.py]: Same as above

File: measurements/api/measurements.py (hunk 1 above)
Comment (FedericoCeratto) [reply measurements/api/measurements.py]: We are treating NULL as an absence of anomaly (hence we coalesce FALSE), so if someone is querying the API using anomaly=False we should return any row that is not TRUE

**Comment by FedericoCeratto** [comment]
rebased

File: measurements/api/measurements.py (hunk 1 above)
Comment (hellais) [reply measurements/api/measurements.py]: Ok fair enough. We are pretty light on details in the spec, but I am pretty sure this is a behaviour change compared to the previous API version.

I am fine doing this, but I figured I would flag it to be sure we this is what we want.

**Pull request merged**